examples/learn/response_models/from_call_args/anthropic/base_message_param.py

```diff
@@ -0,0 +1,18 @@
+from typing import Annotated
+from pydantic import BaseModel, Field
+from mirascope.core import BaseMessageParam, anthropic, FromCallArgs
+
+
+class BookRecommendation(BaseModel):
```

Comment:
we should update all of these examples to show validating an output against the input (because that's really the only purpose of this)

for example:

```python
class Book(BaseModel):
    title: str
    author: str

class Books(BaseModel):
    texts: Annotated[list[str], FromCallArgs()]
    books: list[Book]

    @model_validator(mode="after")
    def validate_output_length(self) -> Self:
        if len(self.texts) != len(self.books):
            raise ValueError("length mismatch...")
        return self

@anthropic.call("claude-3-5-sonnet-20240620", response_model=Books)
def extract_books(texts: list[str]) -> list[BaseMesageParam]:
    return [BaseMessageParam(role="user", content=f"Extract the books from these texts: {texts}")]

texts = [
    "The Name of the Wind by Patrick Rothfuss",
    "Mistborn: The Final Empire by Brandon Sanderson",
]
print(extract_books(texts))
```

this now shows why you would ever care to have the call arguments as part of the response model (since otherwise you could just use the call arguments you passed to the function)